crates/symbolicator-service/src/services/sourcemap_lookup.rs

```diff
@@ -62,6 +64,57 @@ impl SourceMapUrl {
 
 type ArtifactBundle = SelfCell<ByteView<'static>, SourceBundleDebugSession<'static>>;
 
+/// A JS-processing "Module".
+///
+/// This is basically a single file (identified by its `abs_path`), with some additional metadata
+/// about it.
+#[derive(Clone, Debug)]
+pub struct CachedModule {
+    /// The parsed [`Url`] or the original `abs_path` along with a [`url::ParseError`] if it is invalid.
+    abs_path: Result<Url, (String, url::ParseError)>,
+    /// The optional [`DebugId`] of this module.
+    debug_id: Option<DebugId>,
+    // TODO: errors that happened when processing this file
+    /// A flag if have already resolved the minified and sourcemap files.
+    was_fetched: bool,
+    /// The fetched minified JS file.
+    // TODO: this should not be public?
+    pub minified_source: CacheEntry<CachedFile>,
+    /// The converted SourceMap.
+    // TODO: this should not be public?
+    pub smcache: CacheEntry<OwnedSourceMapCache>,
+}
+
+impl CachedModule {
+    fn new(abs_path: &str, debug_id: Option<DebugId>) -> Self {
+        let abs_path = Url::parse(abs_path).map_err(|err| {
+            let error: &dyn std::error::Error = &err;
+            tracing::warn!(error, abs_path, "Invalid Url in JS processing");
+            (abs_path.to_owned(), err)
+        });
+        Self {
+            abs_path,
+            debug_id,
+            was_fetched: false,
+            minified_source: Err(CacheError::NotFound),
+            smcache: Err(CacheError::NotFound),
+        }
+    }
+
+    /// TODO: we should really maintain a list of all the errors that happened for this image?
+    pub fn is_valid(&self) -> bool {
+        self.abs_path.is_ok()
+    }
+
+    /// Creates a new [`FileKey`] for the `file_path` relative to this module
+    pub fn file_key(&self, file_path: &str) -> Option<FileKey> {
+        // FIXME: are the source paths that we symbolicate relative to the minified source or the sourcemap?
```

Comment:
Relative to the sourcemap - https://github.com/getsentry/sentry/blob/b575df197f73f4edadbaec7e4df91d558234ff8f/src/sentry/lang/javascript/processor.py#L1034